pyvista/py.typed

```diff
@@ -0,0 +1 @@
+partial\n
```

Comment:
I have two questions.
1. Could you tell me what `partial` means based on the document?
1. Do we need this file in a subdirectory of pyvista directory?

Reply:
> 1. Could you tell me what partial means based on the document?

`partial` is used to signify that the package is partially type hinted and `mypy` (the static type checker) should search through inline packages and typeshed if modules aren't found with type annotations. Per [PEP 561](https://peps.python.org/pep-0561/#partial-stub-packages):

> Many stub packages will only have part of the type interface for libraries completed, especially initially. For the benefit of type checking and code editors, packages can be “partial”. This means modules not found in the stub package SHOULD be searched for in parts four and five of the module resolution order above, namely inline packages and typeshed.
>
> If a stub package distribution is partial it MUST include `partial\n` in a `py.typed` file. For stub-packages distributing within a namespace package (see [PEP 420](https://peps.python.org/pep-0420)), the `py.typed` file should be in the submodules of the namespace.

2. Do we need this file in a subdirectory of pyvista directory?

No, that would only be for namespace (inline) packages. The `py.typed` file [applies recursively](https://peps.python.org/pep-0561/#partial-stub-packages) for "standard" packages:

> Package maintainers who wish to support type checking of their code MUST add a marker file named py.typed to their package supporting typing. This marker applies recursively: if a top-level package includes it, all its sub-packages MUST support type checking as well. To have this file installed with the package, maintainers can use existing packaging options such as package_data in distutils, shown below.
>
> ```python
> setup(
>     ...,
>     package_data = {
>         'foopkg': ['py.typed'],
>     },
>     ...,
> )
> ```
>
> For namespace packages (see [PEP 420](https://peps.python.org/pep-0420)), the py.typed file should be in the submodules of the namespace, to avoid conflicts and for clarity.

Reply:
In short, for 2, the `py.typed` file should be at the level of the top-level package, which is right inside the `pyvista/pyvista` folder.